internal/awsdata/ecs.go

```diff
@@ -0,0 +1,163 @@
+package awsdata
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/manywho/awsinventory/internal/inventory"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	// AssetTypeECSContainer is the value used in the AssetType field when fetching ECS containers
+	AssetTypeECSContainer string = "ECS Container"
+
+	// ServiceECS is the key for the ECS service
+	ServiceECS string = "ecs"
+)
+
+func (d *AWSData) loadECSContainers(region string) {
+	defer d.wg.Done()
+
+	ecsSvc := d.clients.GetECSClient(region)
+
+	log := d.log.WithFields(logrus.Fields{
+		"region":  region,
+		"service": ServiceECS,
+	})
+	log.Info("loading instance data")
+	var clusterArns []*string
+	done := false
+	params := &ecs.ListClustersInput{}
+	for !done {
+		out, err := ecsSvc.ListClusters(params)
+		if err != nil {
+			d.results <- result{Err: err}
+			return
+		}
+
+		clusterArns = append(clusterArns, out.ClusterArns...)
+
+		if out.NextToken == nil {
+			done = true
+		} else {
+			params.NextToken = out.NextToken
+		}
+	}
+
+	// TODO: API call can only handle 100 cluster ARNs at a time
```

Comment:
For a future update (both for this and DescribeTasks). I think it's probably high enough for now, but can fix it later.

Reply:
I think that's a fair upper limit for now. Like you say, we can always revisit it if it's a problem in the future